core/pta/tests/misc.c

```diff
@@ -551,14 +554,82 @@ static int self_test_nex_malloc(void)
 }
 #endif
 
+#undef LOG
+#define LOG     DMSG_RAW
+
+static int check_virt_to_phys(vaddr_t va, paddr_t exp_pa,
+			      enum teecore_memtypes m)
+{
+	paddr_t pa = 0;
+	void *v = NULL;
+
+	pa = virt_to_phys((void *)va);
+	LOG("virt_to_phys(%#"PRIxVA") => %#"PRIxPA" (expect %#"PRIxPA")",
+	    va, pa, exp_pa);
+	if (pa != exp_pa)
+		goto fail;
+
+	if (!exp_pa)
+		return 0;
+
+	v = phys_to_virt(pa, m, 1);
+	LOG("phys_to_virt(%#"PRIxPA") => %p (expect %#"PRIxVA")",
+	    pa, v, va);
+	if ((vaddr_t)v != va)
+		goto fail;
+	return 0;
+
+fail:
+	LOG("Fail");
+	return -1;
+}
+
+static int self_test_va2pa(void)
+{
+	int ret = 0;
+
+	if (VCORE_FREE_SZ) {
+		vaddr_t va_base = VCORE_FREE_PA;
+		paddr_t pa_base = 0;
+
+		pa_base = virt_to_phys((void *)va_base);
+		if (!pa_base) {
+			LOG("virt_to_phys(%#"PRIxVA") => 0 Fail!", va_base);
+			return -1;
+		}
+
+		/*
+		 * boot_mem_release_unused() and
+		 * boot_mem_release_tmp_alloc() has been called during
+		 * boot.
+		 *
+		 * First pages of VCORE_FREE are expected to be allocated
+		 * with boot_mem_alloc() while the end of VCORE_FREE should
+		 * have been freed by the two mentioned release functions.
+		 */
+		if (check_virt_to_phys(va_base, pa_base, MEM_AREA_TEE_RAM))
+			ret = -1;
+		if (check_virt_to_phys(va_base + 16, pa_base + 16,
+				       MEM_AREA_TEE_RAM))
+			ret = -1;
+		if (check_virt_to_phys(va_base + VCORE_FREE_SZ -
+				       SMALL_PAGE_SIZE, 0, MEM_AREA_TEE_RAM))
```

Comment:
missing `ret = -1` or maybe test to remove?

Reply:
Thanks, I'll add it.